common-lib/src/main/scala/com/gu/mediaservice/lib/aws/ThrallMessageSender.scala

```diff
@@ -1,33 +1,73 @@
 package com.gu.mediaservice.lib.aws
 
-import java.nio.ByteBuffer
-
 import com.gu.mediaservice.lib.config.CommonConfig
 import com.gu.mediaservice.model._
 import com.gu.mediaservice.model.leases.MediaLease
 import com.gu.mediaservice.model.usage.UsageNotice
 import net.logstash.logback.marker.{LogstashMarker, Markers}
 import org.joda.time.DateTime
-import play.api.libs.json.{JodaWrites, Json}
+import play.api.libs.json._
+import play.api.libs.functional.syntax._
 
 import scala.collection.JavaConverters._
 
 // TODO MRB: replace this with the simple Kinesis class once we migrate off SNS
 class ThrallMessageSender(config: CommonConfig) {
   private val kinesis = new Kinesis(config)
 
-  def publish(updateMessage: UpdateMessage): Unit = {
-    kinesis.publish(updateMessage)
-  }
+  def publish(message: ThrallMessage): Unit = kinesis.publish(message)
 }
 
-object UpdateMessage {
-  implicit val yourJodaDateWrites = JodaWrites.JodaDateTimeWrites
-  implicit val unw = Json.writes[UsageNotice]
-  implicit val writes = Json.writes[UpdateMessage]
+object ThrallMessage {
+  implicit val dateWrites = JodaWrites.JodaDateTimeWrites
+  implicit val dateReads = JodaReads.DefaultJodaDateTimeReads
+  implicit val usageReads = Json.reads[UsageNotice]
+  implicit val usageWrites = Json.writes[UsageNotice]
+
+  implicit val messageReads: Reads[ThrallMessage] = {
+    val updateMsg = Json.reads[UpdateMessage]
+    val reindexMsg = Json.reads[ThrallMessage.ReindexImage]
+    __.read[UpdateMessage](updateMsg).map(x => x: UpdateMessage) |
+      __.read[ThrallMessage.ReindexImage](reindexMsg).map(x => x: ThrallMessage.ReindexImage)
+  }
+
+  implicit val messageWrites: Writes[ThrallMessage] = Writes[ThrallMessage] {
+    case msg: UpdateMessage => Json.writes[UpdateMessage].writes(msg)
+    case msg: ThrallMessage.ReindexImage => Json.writes[ThrallMessage.ReindexImage].writes(msg)
+  }
+
+  final case class ReindexImage(id: String, image: String) extends ThrallMessage {
+    override def getId: String = id
+    val subject = "reindex-image"
+  }
 }
 
 // TODO add RequestID
+
+sealed abstract class ThrallMessage extends Product with Serializable {
+  def getId: String
+  val subject: String
+
+  def toLogMarker: LogstashMarker = {
```

Comment:
I think this might be worth a slight revisit soon, as there's places that log id without the marker- and places where we could log a few other markers and it's not easy.